tools/gulp/tasks/aot.ts

```diff
@@ -1,21 +1,21 @@
 import {task} from 'gulp';
+import {copySync} from 'fs-extra';
+import {DIST_DEMOAPP, DIST_RELEASE} from '../constants';
+import {sequenceTask, execNodeTask} from '../util/task_helpers';
 import {join} from 'path';
-import {DIST_ROOT} from '../constants';
-import {execNodeTask, sequenceTask} from '../util/task_helpers';
 
-/** Copies the source files of the demo-app to the dist folder. */
-task('aot:copy', [':build:devapp:scss', ':build:devapp:assets']);
+const tsconfigFile = join(DIST_DEMOAPP, 'tsconfig-aot.json');
 
-/**
- * Prepares the AOT compilation by copying the demo-app and building the components with their
- * associated metadata files from the Angular compiler.
- */
-task('aot:prepare', sequenceTask(
-  'clean',
-  ['aot:copy', 'build:components', ':build:components:ngc'])
-);
+/** Builds the demo-app and library. To be able to run NGC, apply the metadata workaround. */
+task('aot:deps', sequenceTask('build:devapp', ':package:release', 'aot:copy-release'));
 
-/** Builds the demo-app with the Angular compiler to verify that all components work. */
-task('aot:build', ['aot:prepare'], execNodeTask(
-  '@angular/compiler-cli', 'ngc', ['-p', join(DIST_ROOT, 'tsconfig-aot.json')])
-);
+// As a workaround for https://github.com/angular/angular/issues/12249, we need to
+// copy the Material ESM output inside of the demo-app output.
+task('aot:copy-release', () => {
+  copySync(DIST_RELEASE, join(DIST_DEMOAPP, 'material'));
```

Comment:
I tried, but it really felt weird because I would need to specify a glob to match all files. (Just specifying the directory didn't work)

```ts
task('aot:copy-release', () => src(DIST_RELEASE + '**/*').pipe(dest(join(DIST_DEMOAPP, 'material'))));
```

It is definitely a `gulp-y` way, but I think it's just not very clear and also we would need to have a glob.